src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -1075,6 +1111,265 @@ bool FullyConnected::useSparseWeightsDecompression() {
+   So, just passing the `nullptr` is OK.
+ */
+void FullyConnected::primExecLLMFc(void* weight) {
+    // src
+    auto srcPtr = getParentEdgeAt(DATA_ID)->getMemoryPtr();
+    auto* src = srcPtr->getData();
+    auto& data_dims = srcPtr->getStaticDims();
+
+    // dst
+    auto* dst = getChildEdgeAt(0)->getMemoryPtr()->getData();
+
+    // M, N, K
+    auto M = data_dims[0];
+    auto N = weightDims[0];
+    auto K = weightDims[1];
+    if (data_dims.size() == 3) {
+        M *= data_dims[1];
+        assert(K == data_dims[2]);
+    } else {
+        assert(K == data_dims[1]);
+    }
+    auto work_amount = rnd_up(N, 32) / 32;
+    float* bias = nullptr;
+    if (withBiases) {
+        bias = biasRnd.get();
+    }
+    auto thread_num_fcllm = fcLLMs.size();
+    size_t lda, ldb, ldc;
+    lda = K * dnnl::memory::data_type_size(inputDataType);
+    ldb = K * dnnl::memory::data_type_size(inputDataType);
+    ldc = N * dnnl::memory::data_type_size(outputDataType);
+    parallel_for(thread_num_fcllm, [&](size_t tid) {
+        size_t start {0}, end {0};
+        dnnl::impl::balance211(work_amount, thread_num_fcllm, tid, start, end);
+        size_t n0 = start * 32;
+        size_t n1 = std::min(end * 32, N);
+        if (n0 >= N) return;
+
+        llmdnn::fc_kernel_execute(fcLLMs[tid].get(), src, weight, dst, lda, ldb, ldc, M, N, K, n0, n1,
+            dequant.get(), requant.get(), bias);
+    });
+}
+
+bool FullyConnected::tryUseLLMFc() {
+    if (stateLLMFc != Not_Init)
+        return stateLLMFc == State_Use;
+
+    llmdnn::fc_create_param param;
+    if (!tryExtractParamForLLMFc(param)) {
+        stateLLMFc = State_NotUse;
+        return false;
+    }
+
+    auto thread_num = utility::get_total_threads();
+    fcLLMs.resize(thread_num);
+    std::atomic<bool> ret{true};
+    // force to reference the function once
+    utility::simple_parallel_for(thread_num, [&] (size_t i) {
+        llmdnn::fc_kernel* fc;
+        if (!fc_kernel_create(&fc, &param)) {
+            ret = false;
+            return;
+        }
+        fcLLMs[i] = std::shared_ptr<llmdnn::fc_kernel>(fc, [](llmdnn::fc_kernel* p) { fc_kernel_destroy(p); });
+    });
+    if (ret) {
+        stateLLMFc = State_Use;
+        NodeDesc *selected_pd = getSelectedPrimitiveDescriptor();
+        selected_pd->setImplementationType(gemm_llmdnn);
+
+        // If the input precision is fp32 on SPR, the weight will be cast to bf16 for acceleration.
+        auto weight_ptr = tryConvertMemoryPrecision(getParentEdgeAt(WEIGHTS_ID)->getMemoryPtr(), Precision::BF16);
+        weightDims = weight_ptr->getStaticDims();
+        void* weight = weight_ptr->getData();
+        primExecLLMFc(weight);
+    } else {
+        // fallback
+        stateLLMFc = State_NotUse;
+        fcLLMs.clear();
+    }
+
+    return ret;
+}
+
+bool FullyConnected::tryExecLLMFc() {
+    if (stateLLMFc != State_Use)
+        return false;
+
+    void* weight = nullptr;
+    primExecLLMFc(weight);
```

Comment:
why we need pass a `nullptr` to `primExecLLMFc`. Does this implies that the weight is already packed before ?

Reply:
Yes, when creating kernel, the weights is packed, and the weight pointer will be no need in the subsequent calls.

Reply:
A dedicated interface is added for pack, and weight parameter is removed from execLLMFc function.